template_test/Performance_template.py

```diff
@@ -0,0 +1,232 @@
+"""
+Test to <full description of this test file>
+"""
+# Builtin modules
+import logging
+import time
+from uuid import uuid4
+
+# 3rd party modules
+import pytest
+
+# Local modules
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import performance
+from ocs_ci.helpers.helpers import get_full_test_logs_path
+from ocs_ci.ocs.perfresult import PerfResult
+from ocs_ci.ocs.perftests import PASTest
+
+
+log = logging.getLogger(__name__)
+
+
+class ResultsAnalyse(PerfResult):
+    """
+    This class generates results for all tests as one unit
+    and saves them to an elastic search server on the cluster
+
+    """
+
+    def __init__(self, uuid, crd, full_log_path):
+        """
+        Initialize the object by reading some of the data from the CRD file and
+        by connecting to the ES server and read all results from it.
+
+        Args:
+            uuid (str): the unique uid of the test
+            crd (dict): dictionary with test parameters - the test yaml file
+                        that modify it in the test itself.
+            full_log_path (str): the path of the results files to be found
+
+        """
+        super(ResultsAnalyse, self).__init__(uuid, crd)
+
+        # The self.new_index parameter should be unique in the OCS-CI and in the ES
+        self.new_index = "testname_fullres"
+
+        self.full_log_path = full_log_path
+
+        # make sure we have connection to the elastic search server
+        self.es_connect()
+
+    def analyze_the_results(self, test_results):
+        """
+        Analyzing the test results
+
+        Args:
+            test_results (type) : description of argument
+
+        Return:
+            type : description of return value
+        """
+
+        # in this place do the in-test results analyze.
+        # this function is not mandatory.
+
+    def results_logging(self):
+        """
+        Logging relevant results to the test log
+
+        """
+        log.info("Test Results Summery:")
+
+        # Log here the summer of the test results
+
+
+@performance
+class TestClassName(PASTest):
```

Comment:
Maybe it's worth mentioning that TestClassName should be same as test ( file) name

Reply:
the same name of file and test class is not mandatory.